streamingpro-core/src/main/java/tech/mlsql/dsl/adaptor/DslAdaptor.scala

```diff
@@ -77,22 +77,28 @@ trait DslTool {
         split(",").map(_.trim).filterNot(_.isEmpty)
     }
 
-    val currentSchema = new URI(path).getScheme
-    if (filePathSchemas.contains(currentSchema)) {
-      return path
-    }
-
     val newPath = cleanStr(path)
     if (prefix.isEmpty) return newPath
 
-    if (path.contains("..")) {
+    if (newPath.contains("..")) {
       throw new RuntimeException("path should not contains ..")
     }
-    if (path.startsWith("/")) {
-      return prefix + path.substring(1, path.length)
+
+    if (newPath.startsWith("/")) {
+      return prefix + newPath.substring(1, newPath.length)
     }
-    return prefix + newPath
 
+    try {
+      val url = new URL(path)
+      val currentSchema = new URI(url.getProtocol, url.getHost, url.getPath, url.getQuery, null).getScheme
```

Comment:
Is there any different between url.getProtocol and uri.getSchema ?

Reply:
It looks like that the URL only supports http/https related protocal
![image](https://user-images.githubusercontent.com/797758/205270754-e3279da1-9143-4e73-9d54-e8168f589378.png)